utils/appliance/implementations/ssui.py

```diff
@@ -20,22 +23,47 @@ def __init__(self, selenium, endpoint, extra_objects=None):
         })
         super(MiqSSUIBrowser, self).__init__(
             selenium,
-            plugin_class=DefaultPlugin,
+            plugin_class=MiqSSUIBrowserPlugin,
             logger=create_sublogger('MiqSSUIBrowser'),
             extra_objects=extra_objects)
+        self.window_handle = selenium.current_window_handle
 
     @property
     def appliance(self):
         return self.extra_objects['appliance']
 
     def create_view(self, *args, **kwargs):
-        return self.appliance.browser.create_view(*args, **kwargs)
+        return self.appliance.ssui.create_view(*args, **kwargs)
 
     @property
     def product_version(self):
         return self.appliance.version
 
 
+class MiqSSUIBrowserPlugin(DefaultPlugin):
+    ENSURE_PAGE_SAFE = (
+        '''if(typeof $ == 'undefined') { return true; } else { return !($.active > 0); }''')
+
+    DEFAULT_WAIT = 0.8
+
+    def ensure_page_safe(self, timeout='20s'):
+        # THIS ONE SHOULD ALWAYS USE JAVASCRIPT ONLY, NO OTHER SELENIUM INTERACTION
+
+        def _check():
+            result = self.browser.execute_script(self.ENSURE_PAGE_SAFE, silent=False)
+            # TODO: Logging
+            return bool(result)
+
+        wait_for(_check, timeout=timeout, delay=2, silent_failure=False, very_quiet=False)
+
+    def after_keyboard_input(self, element, keyboard_input):
```

Comment:
@mfalesni : Added this method , Tried to implement some OBSERVED_FIELD_MARKERS like in ui but it didn't work . 
Need to implement this .

Reply:
We probably don't need this functionality in here. Maybe just call the ensure_page_safe in this method, since there should be no such observed fields like in the main ui.